gigascience/gigadb-website
# Admin styles - Grid view dataset tables

# Pull request for issue: #1502

This is a pull request for the following functionalities:

* Update styles for all admin pages that look similar to http://gigadb.gigasciencejournal.com:9170/adminDataset/admin

## How to test?

Navigate to all similar pages and cross check that styles have been updated

Here's the full list:

 http://gigadb.gigasciencejournal.com:9170/adminDataset/admin
 http://gigadb.gigasciencejournal.com:9170/adminDatasetAuthor/admin
 http://gigadb.gigasciencejournal.com:9170/adminDatasetSample/admin
 http://gigadb.gigasciencejournal.com:9170/adminFile/admin
 http://gigadb.gigasciencejournal.com:9170/adminDatasetProject/admin
 http://gigadb.gigasciencejournal.com:9170/adminLink/admin
 http://gigadb.gigasciencejournal.com:9170/adminRelation/admin
 http://gigadb.gigasciencejournal.com:9170/datasetFunder/admin
 http://gigadb.gigasciencejournal.com:9170/adminManuscript/admin
 http://gigadb.gigasciencejournal.com:9170/adminAuthor/admin
 http://gigadb.gigasciencejournal.com:9170/adminSample/admin
 http://gigadb.gigasciencejournal.com:9170/adminSpecies/admin
 http://gigadb.gigasciencejournal.com:9170/adminProject/admin
 http://gigadb.gigasciencejournal.com:9170/adminExternalLink/admin
 http://gigadb.gigasciencejournal.com:9170/adminLinkPrefix/admin
 http://gigadb.gigasciencejournal.com:9170/funder/admin
 http://gigadb.gigasciencejournal.com:9170/adminDatasetType/admin
 http://gigadb.gigasciencejournal.com:9170/adminFileType/admin
 http://gigadb.gigasciencejournal.com:9170/adminFileFormat/admin
 http://gigadb.gigasciencejournal.com:9170/news/admin
 http://gigadb.gigasciencejournal.com:9170/rssMessage/admin
 http://gigadb.gigasciencejournal.com:9170/adminPublisher/admin
 http://gigadb.gigasciencejournal.com:9170/datasetLog/admin
 http://gigadb.gigasciencejournal.com:9170/attribute/admin
 http://gigadb.gigasciencejournal.com:9170/user/admin

## How have functionalities been implemented?

The implementation consisted on extending the changes from some of the pages to all of them, and is as follows:

- Use new layout (polyfills are needed for table sorting and filtering)

```php
		$this->layout = 'new_main';
		$this->loadBaBbqPolyfills = true;
```
- wrap page in `<div class="container">`
- Add breadcrumb
```php
<?php
$this->widget('TitleBreadcrumb', [
    'pageTitle' => 'Manage Files',
    'breadcrumbItems' => [
        ['label' => 'Datasets', 'href' => '/site/admin'],
        ['isActive' => true, 'label' => 'Manage'],
    ]
]);
?>
```
- Add message such as this, if it suits page:
```html
<div class="mb-10">
</div>
<p>
  Column headers with links are sortable. Cells with a text input are used for filtering.
</p>
```
- Use `CustomGridView` (for styled pagination)
- Use custom action icons
```php
CustomGridView::getDefaultActionButtonsConfig()
```

## Any issues with implementation?

* I noticed the new layout lacks the https://hypothes.is/ script. Many pages don't currently have this script in the live site, so I'm not sure if this should be included or not, and in which pages >> Script should only be used in dataset/xxxxx pages

## Any changes to automated tests?

It was necessary to do simple updates to a few acceptance tests:

* New layout adds an input so that had to be taken into account in the input indexing
* New text added, so some assertions that did not expect a certain text were removed

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
